pytest_django/plugin.py

```diff
@@ -327,6 +329,86 @@ def restore():
         request.addfinalizer(restore)
 
 
+@pytest.fixture(autouse=True, scope='session')
+def _fail_for_invalid_template_variable(request):
+    """Fixture that fails for invalid variables in templates.
+
+    This fixture will fail each test that uses django template rendering
+    should at template contain an invalid template variable.
+
+    It does not raise an exception, but fails, as the stack trace doesn't
+    offer any helpful information to debug.
+    This behavior can be switched of using the marker:
+    ``ignore_template_errors``
+    """
+    class InvalidVarException(object):
+
+        """Custom handler for invalid strings in templates."""
+
+        def __init__(self):
+            self.fail = True
+
+        def __contains__(self, key):
+            """There is a test for '%s' in TEMPLATE_STRING_IF_INVALID."""
+            return key == '%s'
+
+        def _get_template(self):
+            from django.template import Template
+
+            stack = inspect.stack()
+            # finding the ``render`` needle in the stack
+            frame = reduce(
+                lambda x, y: y[3] == 'render' and 'base.py' in y[1] and y or x,
+                stack
+            )
+            # assert 0, stack
+            frame = frame[0]
+            # finding only the frame locals in all frame members
+            f_locals = reduce(
+                lambda x, y: y[0] == 'f_locals' and y or x,
+                inspect.getmembers(frame)
+            )[1]
+            # ``django.template.base.Template``
+            template = f_locals['self']
+            if isinstance(template, Template):
+                return template
+
+        def __mod__(self, var):
+            """Handle TEMPLATE_STRING_IF_INVALID % var."""
+            template = self._get_template()
+            if template:
+                msg = "Undefined template variable '%s' in '%s'" % (var, template.name)
+            else:
+                msg = "Undefined template variable '%s'" % var
+            if self.fail:
+                pytest.fail(msg, pytrace=False)
+            else:
+                return msg
+
+    if django_settings_is_configured():
```

Comment:
I can't the problem is, that in the first time I set the setting. Than all template settings get copied into to engine. That's why you can't change the template setting during runtime.
That's why in the second very similar looking snippet, I use the hashable quality of an object to change it state. Which also changes the engines own store as well.

I could define a singleton tho. Than I would only need to change the state in the marker. I'll look into that.